crates/iceberg/src/arrow/reader.rs

```diff
@@ -523,6 +568,93 @@ impl ArrowReader {
         Ok(results.into())
     }
 
+    /// Helper function to add a `_file` column to a RecordBatch at a specific position.
+    /// Takes the array, field to add, and position where to insert.
+    fn create_file_field_at_position(
```

Comment:
I think this approach is not extensible. I prefer what's similar in [this pr](https://github.com/apache/iceberg-rust/pull/1821/files#diff-640415fbb0b37eac6f774eff524fb364c4e54c02da3a3c0e355af5caa941007c):
1. Add `constant_map` for `ArrowReader`
2. Add another variant of `RecordBatchTransformer` to handle constant field like `_file`

Reply:
I sketched an approach in the record batch transformer, I took the path of the transformer just having a `constant_map` stored which can be populated by the reader.